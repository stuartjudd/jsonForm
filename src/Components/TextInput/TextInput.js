import React, { useState} from "react"

const TextInput = ({id, name, label, classWrapper, placeholder, validation, required}) => {
    
    const [inputValue, setInputValue] = useState('')
    const [inputValueError, setInputValueError] = useState('')

    const onBlur = () =>
    {
       if (validation){ 
        return validation.map( (rule) => {
            switch (rule.type){
                case "regex":                           
                    const regExp = new RegExp(rule.expression)
                    if (!regExp.test(inputValue)){
                         setInputValueError(rule.message)
                    }
                    break;
                case "required" :
                    if (inputValue.length < 1 )
                         setInputValueError(rule.message)
                    break;
                
                case "age" :  
                    const currentDate = new Date()
                    var dateParts = inputValue.split("/");                    
                    var regDate = new Date(+dateParts[2]+ parseInt(rule.low) , dateParts[1] - 1, +dateParts[0] ) ;                                                          
                    if (currentDate < regDate  )
                        setInputValueError(rule.message)
               break;
                default :
                    break                    
            }
        })           
       }       
    }
    return (
        <div className={classWrapper}>
            <label htmlFor={id}>{label} </label>
            <input 
                id={id}
                type="input"               
                name={name}
                placeholder={placeholder}
                value = {inputValue}
                required = { required ? true : false}
                onChange= {event => setInputValue(event.target.value)}
                onBlur={onBlur}
                onFocus={event => setInputValueError("")}                             
               ></input> 
               <br/>                
                {inputValueError ?  <span className='error'>{inputValueError}</span> : "" }                           
        </div>        
    )
}

export default TextInput;