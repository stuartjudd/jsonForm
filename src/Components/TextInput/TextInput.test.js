import React from 'react'
import {shallow} from 'enzyme'
import TextInput from './TextInput'

// Start Testing Inputs

describe('<TextInput> component',() => {
    it('should render', () => {
        const component = shallow(<TextInput />)
        expect(component).toMatchSnapshot();            
    })
})

describe('<TextInput> component',() => {
    const name = '123'
    it('should render', () => {
        const component = shallow(<TextInput name={name}/>)
        expect(component).toMatchSnapshot();            
    })
})







