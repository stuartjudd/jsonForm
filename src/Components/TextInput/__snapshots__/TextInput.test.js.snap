// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput> component should render 1`] = `ShallowWrapper {}`;

exports[`<TextInput> component should render 2`] = `ShallowWrapper {}`;
